app/controllers/repositories_controller.rb

```diff
@@ -224,15 +223,8 @@ def create_equipments
       end if params['equipments'].present?
     end
 
-    def update_share_type
-      if repository_params['share_type'].present?
-        @repository.share_type = repository_params['share_type']
-        @repository.save
-      end
-    end
-
     def update_password
-      if @repository.share_type.eql?("public")
+      if repository_params['share_type'].eql?("public")
```

Comment:
@tahnok I modified the conditions so I could get rid of the update_share_type method.
Does this look better?
The update_password method is necessary in cases where the repo changes from private to public, I need the password to be set to nil manually.